lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_producer.rb

```diff
@@ -0,0 +1,354 @@
+          end
+        end
+
+        def columns
+          {
+            distributor: :distributor,
+            producer: :producer,
+            producer_tax_status: :producer_tax_status,
+            order_cycle: :order_cycle,
+            enterprise_fee_name: :enterprise_fee_name,
+            enterprise_fee_type: :enterprise_fee_type,
+            enterprise_fee_owner: :enterprise_fee_owner,
+            tax_category: :tax_category,
+            tax_rate_name: :tax_rate_name,
+            tax_rate: :tax_rate_amount,
+            total_excl_tax: :total_excl_tax,
+            tax: :tax,
+            total_incl_tax: :total_incl_tax
+          }
+        end
+
+        def rules
+          [
+            { group_by: :distributor },
+            { group_by: :producer },
+            {
+              group_by: :order_cycle,
+              summary_row: proc do |_key, items, _rows|
+                order_ids = items.flat_map(&:second).map(&:id).uniq
+                included_tax = included_tax_for_order_ids(order_ids)
+                tax = tax_for_order_ids(order_ids)
+                total_excl_tax = enterprise_fees_amount_for_orders(order_ids) - included_tax
+                {
+                  total_excl_tax: total_excl_tax,
+                  tax: tax,
+                  total_incl_tax: total_excl_tax + tax
+                }
+              end
+            }
+          ]
+        end
+
+        def included_tax_for_order_ids(order_ids)
+          Spree::Adjustment.tax
+            .where(order: order_ids)
+            .where(included: true)
+            .where(adjustable_type: 'Spree::Adjustment')
+            .where(adjustable_id: enterprise_fee_adjustment_ids_for_orders(order_ids))
+            .pluck("sum(amount)")
+            .first || 0
+        end
+
+        def tax_for_order_ids(order_ids)
+          Spree::Adjustment.tax
+            .where(order: order_ids)
+            .where(adjustable_type: 'Spree::Adjustment')
+            .where(adjustable_id: enterprise_fee_adjustment_ids_for_orders(order_ids))
+            .pluck("sum(amount)")
+            .first || 0
+        end
+
+        def enterprise_fee_adjustment_ids_for_orders(order_ids)
+          enterprise_fees_for_orders(order_ids).pluck(:id)
+        end
+
+        def enterprise_fees_amount_for_orders(order_ids)
+          enterprise_fees_for_orders(order_ids).pluck("sum(amount)").first || 0
+        end
+
+        def enterprise_fees_for_orders(order_ids)
+          enterprise_fees = Spree::Adjustment.enterprise_fee
+            .where(order_id: order_ids)
+          return enterprise_fees unless enterpise_fee_filters?
+
+          enterprise_fees.where(
+            originator_id: enterprise_fee_filtered_ids
+          )
+        end
+
+        def distributor(query_result_row)
+          first_order(query_result_row).distributor&.name
+        end
+
+        def producer(query_result_row)
+          Enterprise.where(id: supplier_id(query_result_row)).pluck(:name).first
```

Comment:
I just learned about `pick` which applies `limit 1` as well:
```suggestion
          Enterprise.where(id: supplier_id(query_result_row)).pick(:name)
```